### ISSUE TITLE
error in combining of regular epressions

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#3760]
Migrated from [rt.perl.org#6650](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=6650) (status was 'resolved')

Searchable as RT6650$

**Comment by p5pRT** [comment]
### From hans@kolej.mff.cuni.cz



<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00503:

Configured by drow at Sun Apr 30 12:07:23 EDT 2000.

Summary of my perl5 (5.0 patchlevel 5 subversion 3) configuration:
  Platform:
    osname=linux, osvers=2.2.15pre14, archname=i386-linux
    uname='linux them 2.2.15pre14 #2 smp mon mar 13 14:29:00 est 2000 i686 unknown '
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='cc', optimize='-O2 ', gccversion=2.95.2 20000313 (Debian GNU/Linux)
    cppflags='-Dbool=char -DHAS_BOOL -D_REENTRANT -DDEBIAN -I/usr/local/include'
    ccflags ='-Dbool=char -DHAS_BOOL -D_REENTRANT -DDEBIAN -I/usr/local/include'
    stdchar='char', d_stdstdio=undef, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldbm -ldb -ldl -lm -lc -lposix -lcrypt
    libc=, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl 5.00503:
    /usr/lib/perl5/5.005/i386-linux
    /usr/lib/perl5/5.005
    /usr/local/lib/site_perl/i386-linux
    /usr/local/lib/site_perl
    /usr/lib/perl5
    .


Environment for perl 5.00503:
    HOME=/home/hans
    LANG=cs_CZ.ISO-8859-2
    LANGUAGE (unset)
    LC_COLLATE=cs_CZ
    LC_CTYPE=cs_CZ
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=~/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @vanstyn

Hi\, thanks for the report\. Unfortunately none of the strings I've tried
fail under v5\.005\_03 on my Redhat system here&#8203;: could you please give one
or more examples of specific strings that fail?

Hugo van der Sanden

**Comment by p5pRT** [comment]
### From @vanstyn

Ah\, I understand now why I couldn't reproduce it&#8203;: this bug has already
been patched in the 5\.005\_03 I've installed here\.

The problem occurs because flags on embedded regexps leak out into the
next character of the embedding regexp\, so that when the outer regexp
has /x and the embedded regexp does not\, the space or tab immediately
following the embedded regexp is literally required\.

Workarounds&#8203;: match the flags\, or embed only literals rather than
pre\-compiled qr\{\} regexps\. Fix&#8203;: use a later perl\, or apply the patch at
http&#8203;://www\.xray\.mpe\.mpg\.de/mailing\-lists/perl5\-porters/1999\-07/msg00004\.html
\(also bugid 19990630\.004\)\.

Hugo
